e2e/transport.go

```diff
@@ -46,6 +47,37 @@ type transportHook interface {
 	Gossip(from, to pbft.NodeID, msg *pbft.MessageReq) bool
 }
 
+// encapsulates routing mapping for certain round
+type roundMetadata struct {
+	round uint64
+	// represents message routing map (sender node id to recipient node ids)
+	// e.g. A4 sends messages to [A0, A1]
+	routingMap map[pbft.NodeID][]pbft.NodeID
+}
+
+// callback which enables determining which message should be gossiped
+type gossipArbitrageHandler func(sender, reciever pbft.NodeID, msg *pbft.MessageReq) (shouldSend bool)
```

Comment:
Why does the type contain the structure `arbitrage`?

Reply:
First slice of implementation implied that only specific part for this test is contained within the delegate and if delegate doesn't manage to handle it, in that case messages are routed based on provided `routingMap` within the `Gossip` function itself. However this is now obsolete, since all the gossip logic is contained within the callback, therefore you are right, we will remove arbitrage word from it, so it will be called `gossipHandler`.